packages/components/src/summary/number.js

```diff
@@ -144,6 +146,10 @@ SummaryNumber.propTypes = {
 	 * A string or number value to display - a string is allowed so we can accept currency formatting.
 	 */
 	value: PropTypes.oneOfType( [ PropTypes.number, PropTypes.string ] ),
+	/**
+	 * A function to be called after a SummaryNumber, rendered as a link, is clicked.
+	 */
+	onLinkCallback: PropTypes.func,
```

Comment:
I'm open to naming suggestions here. `onToggle` is used to determine if component is a link or button and I'd like to provide a way for a function to be called after the link has been activated.